lib/opensusebasetest.pm

```diff
@@ -613,7 +613,8 @@ sub wait_boot {
         }
     }
     # On Xen PV and svirt we don't see a Grub menu
-    elsif (!(check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux') && check_var('BACKEND', 'svirt'))) {
+    # If KEEP_GRUB_TIMEOUT is defined it means that GRUB menu will appear only for one second
+    elsif (!(check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux') && check_var('BACKEND', 'svirt') || check_var('KEEP_GRUB_TIMEOUT', '1'))) {
```

Comment:
Why not `get_var('KEEP_GRUB_TIMEOUT')` instead of `check_var`?

Reply:
yeah sure you can , but what will be benefit in comparison to check_var ? I prefer more strict way to check it via check_var ...

Reply:
@asmorodskyi "more strict way" is not a good reason. Just simply don't ever check for equality on booleans, plz.

Reply:
> @asmorodskyi "more strict way" is not a good reason. Just simply don't ever check for equality on booleans, plz.

as far as I know perl does not have boolean type so can you please be more specific with your request ?  
Regarding "more strict way" - I don't want function to pass in someone will put `KEEP_GRUB_TIMEOUT=no` or `KEEP_GRUB_TIMEOUT=false` or even if someone "correctly" put there  `KEEP_GRUB_TIMEOUT=true` . If you take a look in osd you will see that in most cases people using '0/1'. So by using `check_var('KEEP_GRUB_TIMEOUT','1');` I making sure that my code is follow this approach which I found a right way to go.  Can you please bring some practical reasons why it is bad because for now you just said that you don't like it ?

Reply:
well, that certainly is a specific aspect of the perl language design. What you are describing is mixing input syntax validation with value check, not very common at least, I guess :) Using "1" as the only allowed value is just a convention. https://perldoc.perl.org/perldata.html#Scalar-values describes what you also mentioned: Many values would evaluate to true or false. I think we have way more common use of `get_var` to evaluate a variable in a boolean context than what you propose. To simply have a consistent style I suggest we avoid `check_var(…, 1)` but that's just it: A suggestion, nothing else.